### PR TITLE
Map ompl's APPROXIMATE_SOLUTION -> TIMED_OUT / PLANNING_FAILED

### PR DESCRIPTION
**Pull request by rhaschke**
ompl's `APPROXIMATE_SOLUTION` is not suitable for actual execution. It just states that we [got closer to the goal](https://ompl.kavrakilab.org/classompl_1_1base_1_1ProblemDefinition.html#a612902cacc3b1a2bcc367a9e82739132):

> If the top found solution is approximate, does not actually reach the desired goal, but hopefully is closer to it

As the most prominent reason for an approximate solution is a timeout, also checking for this case.

Fixes https://github.com/ros-planning/moveit_task_constructor/issues/485

### CODE REVIEW

**Comment by rhaschke** [comment]
Interestingly the issue was fixed in the MoveIt1 backport (https://github.com/ros-planning/moveit/pull/3257#pullrequestreview-1163396768), but then not back-ported to MoveIt2 again!

File: moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp
Comment (Abishalini) [inline moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp:1006]: Nit: Can we print the planning timeout value here along with the `Planning timed out` message?
```suggestion
        RCLCPP_WARN(LOGGER, "Planning timed out. Allowed planning time is currently set to %f", request_.allowed_planning_time
```

**Review by Abishalini — APPROVED** [review]
Thanks for this fix! 
I tested with a setup that often executed partial solutions and this fixes it. I had to increase the planning time in my setup.

**Review by henningkayser — APPROVED** [review]

**Comment by leander2189** [comment]
Thanks for the fix!!

What is the procedure to get this PR merged also into **humble** branch? I recall there is a bot that does that, but have no idea how to invoke it...

**Comment by rhaschke** [comment]
@mergifyio backport humble

**Comment by rhaschke** [comment]
> What is the procedure to backport? I recall there is a bot that does that, but have no idea how to invoke it...

Usually, it's done by providing the label `backport-humble` (before merging), which I forgot to do...
However, one can manually trigger a backport by providing a comment (see https://github.com/ros-planning/moveit2/pull/2455#issuecomment-1768315627). I will merge the backport if CI succeeds.

Note, that we will need a new release to actually have these changes deployed.

**Comment by leander2189** [comment]
> Note, that we will need a new release to actually have these changes deployed.

Yes, I understand. Thank you

**Comment by leander2189** [comment]
Just to note that these changes are still not deployed in latest binary version from apt repos (2.5.5-1jammy.20240217.122953).

**Pull request merged**
